### ISSUE TITLE
<incompatibilities> in module.manifest seems not working correctly

### ISSUE THREAD

**Issue by leelu611** [VirtoCommerce/vc-platform#1097]
Hi there,

We recently included the latest version of VirtoCommerce.Search and VirtoCommerce.SearchApi moduels into our project, and we found that Virto Commerce is throwing the exception that saying the VirtoCommerce.Core module is incompatible with them.

After some investigation, we think the issue might be in the ManifestModuleCatalog.Validate() method:

                    var installedIncompatibilities = modules.Select(x => x.Identity).Join(module.Incompatibilities, x => x.Id, y => y.Id, (x, y) => new { x, y })
                                                            .**Where(g => g.y.Version.IsCompatibleWith(g.x.Version))**.Select(g => g.x)
                                                            .ToArray();

The highlighted line above seems to be doing in the opposite way. Please review and check.

Version info: 
- Browser version: Google Chrome
- Platform version: 2.13.10
- Module version: VirtoCommerce.Core - 2.24.2, VirtoCommerce.Search - 2.14.4, VirtoCommerce.SearchApi - 1.3.6

### Expected behavior
VirtoCommerce.Core 2.24.2 should be compatible with the installed modules VirtoCommerce.Search 2.14.4 and VirtoCommerce.SearchApi 1.3.6

### Actual behavior
VirtoCommerce.Core 2.24.2 is incompatible with installed VirtoCommerce.Search 2.14.4, VirtoCommerce.SearchApi 1.3.6. You should uninstall these modules first.

### Steps to reproduce
1. Install version 2.24.2 of VirtoCommerce.Core
2. Install version 2.14.4 of VirtoCommerce.Search
3. Install version 1.3.6 of VirtoCommerce.SearchApi

**Comment by Woland2k** [comment]
I believe it is not compatible with version 2 of VirtoCommerce.Search and version 1 of VirtoCommerce.SearchApi. These two libraries has been depreciated and no longer used. You should install specific providers "ElasticSearch", "Lucene" or "AzureSearch".

**Comment by leelu611** [comment]
@Woland2k , thanks for the clarification. We've also finger out that these two modules might be depreciated already, but it is not stated in the module or wiki. And we've already installed the specific search providers.
Anyway, the validate() does take much time for us to understand. And we found that some of the code are duplicate with that from ModuleInstaller.Install(), which might can be refactor a bit.
